state/protocol/protocol_state/epochs/happy_path_statemachine.go

```diff
@@ -98,6 +104,7 @@ func (u *HappyPathStateMachine) ProcessEpochSetup(epochSetup *flow.EpochSetup) (
 	activeIdentitiesLookup := u.parentState.CurrentEpoch.ActiveIdentities.Lookup() // lookup NodeID → DynamicIdentityEntry for nodes _active_ in the current epoch
 	nextEpochActiveIdentities, err := buildNextEpochActiveParticipants(activeIdentitiesLookup, u.parentState.CurrentEpochSetup, epochSetup)
 	if err != nil {
```

Comment:
similarly here: still ambiguity in the error specification (are really only `InvalidServiceEventError` possible?)